src/test/FuzzerImpl.cpp

```diff
@@ -688,17 +763,87 @@ TransactionFuzzer::initialize()
         genOffersForPair(ASSET_B, ASSET_C, {13, 2, 14, 3, 15}, ltx);
         genOffersForPair(ASSET_C, ASSET_D, {6, 3, 7, 4, 8}, ltx);
 
-        // construct transaction
-        auto tx = createFuzzTransactionFrame(mSourceAccountID, ops,
-                                             mApp->getNetworkID());
+        attemptToApplyOps(ltx, mSourceAccountID, ops, *mApp);
+
+        ltx.commit();
+    }
+
+    {
+        LedgerTxn ltx(ltxOuter);
 
-        tx->attemptApplication(*mApp, ltx);
+        xdr::xvector<Operation> ops;
 
-        if (tx->getResultCode() != txSUCCESS)
+        // Reduce account balances so that fuzzing has a better change
+        // of exercising limits.
+        for (uint8_t i = 0; i < FuzzUtils::NUMBER_OF_PREGENERATED_ACCOUNTS; ++i)
         {
-            throw std::runtime_error("Error while initializing ledger state");
+            PublicKey account;
+            FuzzUtils::setShortKey(account, i);
+
+            // Reduce "account"'s native balance by paying the root.
+            LedgerKey accountKey;
+            accountKey.type(ACCOUNT);
+            accountKey.account().accountID = account;
+            LedgerEntry accountEntry = ltx.load(accountKey).current();
+            auto minBalance = getMinBalance(
+                mApp->getLedgerManager().getLastClosedLedgerHeader().header,
+                accountEntry.data.account());
+            auto remainingBalance =
+                minBalance +
+                FuzzUtils::DEFAULT_ASSET_AVAILABLE_FOR_TEST_ACTIVITY +
+                FuzzUtils::FUZZING_FEE *
+                    FuzzUtils::DEFAULT_NUM_TRANSACTIONS_TO_RESERVE_FEES_FOR;
+            assert(FuzzUtils::INITIAL_ASSET_DISTRIBUTION >= remainingBalance);
+            auto reduceNativeBalanceOp = txtest::payment(
```

Comment:
I don't see how this is going to produce the right balance for account 0, because it has been paying fees for all of these set-up transactions. You probably shouldn't be using `INITIAL_ASSET_DISTRIBUTION` here.

Conceptually, why are we using one of the generated source accounts for the set-up transactions instead of using the root account, which we aren't interested in?

Reply:
`getAvailableBalance` will probably help you get this section right.

Reply:
> You probably shouldn't be using INITIAL_ASSET_DISTRIBUTION here.

> getAvailableBalance will probably help you get this section right.

Indeed, using `getAvailableBalance()` got rid of the use of `INITIAL_ASSET_DISTRIBUTION`.

Reply:
> Conceptually, why are we using one of the generated source accounts for the set-up transactions instead of using the root account, which we aren't interested in?

I don't know; I've wondered myself.  It's been that way since day one.  I'd be more than happy to change it, but I feel like that would be confusing to do together with the other changes in this PR; I'd rather make a separate PR for it once this one's in.  What do you think?

Reply:
Better to do it in a separate PR.

Reply:
> I don't see how this is going to produce the right balance for account 0, because it has been paying fees for all of these set-up transactions.

In fact, as long as we're still using account 0 as the source, it's meaningless to reduce account 0's balance at all -- it's just sending a payment to itself.  So in this PR, I'll just start the balance-reducing loop with account `1`.

In the PR that changes the source account to `root`, we can stop special-casing account `0`.

Does that make sense?